python_modules/dagster/dagster/_cli/api.py

```diff
@@ -613,11 +664,18 @@ def grpc_command(
             package_name=kwargs["package_name"],
         )
 
-    with (
-        mock_system_timezone(override_system_timezone)
-        if override_system_timezone
-        else seven.nullcontext()
-    ):
+    with ExitStack() as exit_stack:
+
+        if override_system_timezone:
+            exit_stack.enter_context(mock_system_timezone(override_system_timezone))
+
+        instance = None
+        if load_secrets:
+            with DagsterInstance.from_ref(
+                instance_ref
+            ) if instance_ref else DagsterInstance.get() as instance:
+                instance.inject_secrets_into_environment(location_name)
+
```

Comment:
nit: this is easy to gloss over but pretty fundamental behavior divergence, might be good to add some comments and make it less compact